java/src/jmri/implementation/MatrixSignalMast.java

```diff
@@ -0,0 +1,444 @@
+        outputlist.add(outputsToBeans.get("output1").getName()); // convert NBH to name (String)
+        if (outputsToBeans.containsKey("output2")) {
+            // outputsToBeans hashmap
+            outputlist.add(outputsToBeans.get("output2").getName());
+        }
+        if (outputsToBeans.containsKey("output3")) {
+            outputlist.add(outputsToBeans.get("output3").getName());
+        }
+        if (outputsToBeans.containsKey("output4")) {
+            outputlist.add(outputsToBeans.get("output4").getName());
+        }
+        if (outputsToBeans.containsKey("output5")) {
+            outputlist.add(outputsToBeans.get("output5").getName());
+        }
+        return outputlist;
+    }
+
+    HashMap<String, NamedBeanHandle<Turnout>> outputsToBeans = new HashMap<String, NamedBeanHandle<Turnout>>(5); // output# - bean pairs
+
+    //looks a lot like the next method, remove?
+    NamedBeanHandle<Turnout> TurnoutNameToHandle (String turnoutName) {
+        if (turnoutName != null && !turnoutName.equals("")) {
+            Turnout turn = jmri.InstanceManager.turnoutManagerInstance().getTurnout(turnoutName);
+            NamedBeanHandle<Turnout> namedTurnout = jmri.InstanceManager.getDefault(jmri.NamedBeanHandleManager.class).getNamedBeanHandle(turnoutName, turn);
+            return namedTurnout;
+        }
+        log.error("Trying to create an empty output");
+        return null;
+    }
+
+    /**
+     * Receive properties from xml, convert name to NamedBeanHandle, store in hashmap outputsToBeans
+     * @param colname String describing the name of the corresponding output, like "output1"
+     * @param turnoutname String for the display name of the output, like "LT1"
+    */
+    public void setOutput(String colname, String turnoutname) {
+        if (colname == null || turnoutname == null){
+            log.error("Trying to set a null output. Fix output configuration for mast");
+        } else {
+            Turnout turn = jmri.InstanceManager.turnoutManagerInstance().getTurnout(turnoutname);
+            NamedBeanHandle<Turnout> namedTurnout = jmri.InstanceManager.getDefault(jmri.NamedBeanHandleManager.class).getNamedBeanHandle(turnoutname, turn);
+            if (outputsToBeans.containsKey(colname)) {
+                log.debug("Output " + colname + " is already defined so will override");
+                outputsToBeans.remove(colname);
+            }
+            outputsToBeans.put(colname, namedTurnout);
+        }
+    }
+
+    /**
+     *  Send hardware instruction
+     *  @param bits char[] of on/off outputs per aspect, like "00010"
+     *  length of array should match the number of outputs defined
+     */
+    public void updateOutputs (char[] bits) {
+        if (bits == null){
+            log.debug("Empty char[] received");
+        }
+        for (int i = 0; i < BitNum; i++) {
+            //if (bits[i] == '1' && getTurnoutBean(i).getCommandedState() == Turnout.THROWN) { // no need to set a state already set
```

Comment:
At this point the program fails to interact with the Turnouts, so had to comment these 4 lines for the moment